lib/cache/expirations/Expiration.ts

```diff
@@ -0,0 +1,3 @@
+export interface Expiration {
+  add(key: string, clear: () => any): void;
```

Comment:
Why return `any`? is it used further?
At the same time, would be nice if clear would be called with the `key`.